### ISSUE TITLE
Upgrading Numpy should also update packages depending on it

### ISSUE THREAD

**Issue by astrofrog** [conda/conda#1841]
At the moment, updating Numpy can cause package dependencies to end up in an inconsistent state:

```
$ conda create -n testdeps2 python=2.7
Fetching package metadata: ....
Solving package specifications: .
...
$ source activate testdeps2
discarding /Users/tom/miniconda3/envs/production/bin from PATH
prepending /Users/tom/miniconda3/envs/testdeps2/bin to PATH
$ conda install astropy=1.0.4
Fetching package metadata: ....
Solving package specifications: .
Package plan for installation in environment /Users/tom/miniconda3/envs/testdeps2:

The following NEW packages will be INSTALLED:

    astropy: 1.0.4-np19py27_0
    numpy:   1.9.3-py27_0    

Linking packages ...
[      COMPLETE      ]|###################################################| 100%
$ conda install numpy
Fetching package metadata: ....
Solving package specifications: .
Package plan for installation in environment /Users/tom/miniconda3/envs/testdeps2:

The following packages will be UPDATED:

    numpy: 1.9.3-py27_0 --> 1.10.1-py27_0

Unlinking packages ...
[      COMPLETE      ]|###################################################| 100%
Linking packages ...
[      COMPLETE      ]|###################################################| 100%
mac-robitaille2:tmp tom$ conda list
# packages in environment at /Users/tom/miniconda3/envs/testdeps2:
#
astropy                   1.0.4                np19py27_0  
numpy                     1.10.1                   py27_0  
openssl                   1.0.2d                        0  
pip                       7.1.2                    py27_0  
python                    2.7.10                        2  
readline                  6.2                           2  
setuptools                18.4                     py27_0  
sqlite                    3.8.4.1                       1  
tk                        8.5.18                        0  
wheel                     0.26.0                   py27_1  
zlib                      1.2.8                         0  
```

The issue here is that Astropy is still the one built against Numpy 1.9, so when using some modules which were compiled against the Numpy C API, one gets warnings along the lines of:

```
module compiled against API version a but this version of numpy is 9
```

TL;DR: if updating Numpy, packages that depend on a specific version of Numpy should be updated too.

**Comment by jorisvandenbossche** [comment]
I think I reported something similar here: https://github.com/ContinuumIO/anaconda-issues/issues/493, where in this case a downgrade of numpy did not update pandas which depended on a specific numpy version

**Comment by megies** [comment]
I think this is fixed in conda 4.x and can be closed?

**Comment by mcg1969** [comment]
Yes, you're right. Thanks for the find!
